presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java

```diff
@@ -750,6 +750,20 @@ public void testComparison()
         assertFunction("ARRAY [ARRAY [1, 2], ARRAY [3, 4, 5]] < ARRAY [ARRAY [1, 2], ARRAY [3, 4, 5]]", BOOLEAN, false);
 
         assertInvalidFunction("ARRAY [1, NULL] = ARRAY [1, 2]", NOT_SUPPORTED.toErrorCode());
+
+        assertFunction("ARRAY [NULL] IS DISTINCT FROM ARRAY [NULL]", BOOLEAN, false);
```

Comment:
Make sure you cover all 4 cases in all tests. (null, non-null) x (null, non-null). When both side are null, you'll want to add casts so that it hits the function for the right type (instead of the unknown implementation).

Do the same for all tests.